### ISSUE TITLE
SONIC: Probe finds NDS file, but game instance fails indexing it

### ISSUE THREAD

**Issue by DartPower** [xoreos/xoreos#569]
I can't launch Sonic Chronicles: The Dark Brotherhood

OS: Windows 10 (20H2,19042.928)
Versions of xoreos: 0.0.6 "Elanee"

Problem:
ERROR: No such archive file "F:/Games/XOREOS/Sonic/Sonic.nds"

Log:
```
F:\Games\XOREOS\xoreos-0.0.6-win64>xoreos sonic
Target "sonic"
Found Max FSAA: 0x
OpenGL context successfully created:
OpenGL version: 2.1
FSAA level    : 0x
Initialising shaders...
shader default/text.vert loaded
shader default/text.frag loaded
shader default/texture.vert loaded
shader default/texture.frag loaded
shader default/textureMatrix.vert loaded
shader default/default.vert loaded
shader default/colour.frag loaded
Initialising default surfaces...
Initialising default materials...
Initialising default mesh containers...
Graphics subsystem initialized
Sound subsystem initialized
Event subsystem initialized
Detected game "Sonic Chronicles: The Dark Brotherhood"
[  0%] Declare languages (0.00s)
Available languages:
- English
- French
- German
- Italian
- Spanish
Using the first available language (English)
Setting the language to English
[ 14%] Loading user game config (0.16s)
[ 28%] Indexing the ROM file (0.32s)
ERROR: No such archive file "F:/Games/XOREOS/Sonic/Sonic.nds"
Shutting down
Cleaning up shaders...
```

**Comment by DrMcCoy** [comment]
Hmm, okay, this is weird. The probe does detect the game (and for that, it has to read the NDS file), but then the game instance fails to index it afterwards.

It works no problem on Linux. But I do also see the exact same problem with Wine, so I have a way to test and trace this. Must be some Windows path shenanigans going on there.

I'll see if I can find some time for that on Saturday.

**Comment by DrMcCoy** [comment]
Oh, this was already fixed with #560, commit e5189a8258d8955b11b3252f44089e9ef1637a54. I completely forgot about that one :)

Closing this issue, then.

I'm attaching Windows packages of xoreos 0.0.6+122.gd181895a4, i.e. a compilation from the current main branch: [xoreos-0.0.6-gd181895a4-win32.zip](https://github.com/xoreos/xoreos/files/6409568/xoreos-0.0.6-gd181895a4-win32.zip) , [xoreos-0.0.6-gd181895a4-win64.zip](https://github.com/xoreos/xoreos/files/6409570/xoreos-0.0.6-gd181895a4-win64.zip)

Keep in mind, though, that Sonic isn't at all playable in xoreos right now. You'll only be able to fly-scroll through the starting area, with some of the items showing.

If you want to contribute to xoreos, we'd be happy to have you. Have a look over [the Developer Central page on our wiki](https://wiki.xoreos.org/index.php?title=Developer_Central), which includes some pointers on how to compile xoreos, a general overview of the codebase, a rambly TODO, etc.

Some specific points of interest for Sonic in particular are, in no particular order:
- The Actimagine VX video codec. ffmpeg/libav's Kostya wrote a decoder in Rust, which can be used as a starting point to write our own implementation in C++ for xoreos
- Menus. They description is kept in GFF files with the .gui extension, and the "actions" seem to be some kind state machine kept in GDA files (a table structure)
- Player characters and NPCs. No idea at all how these work yet
- Music. Looks to be some kind of tracker or MIDI format, with instruments in sound_data.sadl file. Dunno if that's common for Nintendo DS games

Though basically anything furthering progress is good. A lot of information on *how* things work in the game isn't yet known to us. Investigating this and adding the information to, for example, our wiki would also be a good starting point. Hit me up on #xoreos on Freenode IRC for an account on our wiki for that, if you want.
